src/client.js

```diff
@@ -2148,7 +2149,12 @@ MatrixClient.prototype.sendReceipt = function(event, receiptType, callback) {
  * @return {module:client.Promise} Resolves: TODO
  * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
-MatrixClient.prototype.sendReadReceipt = function(event, callback) {
+MatrixClient.prototype.sendReadReceipt = async function(event, callback) {
+    const eventId = event.getId();
+    const room = this.getRoom(event.getRoomId());
+    if (room && room.hasPendingEvent(eventId)) {
+        throw new Error(`Cannot set read receipt to a pending event (${eventId})`);
```

Comment:
should we log instead of throwing?

Reply:
I think it's important to throw in these cases so that the [calling code](https://github.com/matrix-org/matrix-react-sdk/pull/3184/files#diff-9dd3a0bf58495df3165ed62d2237e0eeR709) knows to revert its own state and retry again later.

Reply:
ah, fair.